src/test/util/failpoint.rs

```diff
@@ -54,21 +61,39 @@ pub struct FailPointGuard {
     failpoint_name: String,
 }
 
+impl FailPointGuard {
+    pub async fn disable(&self) {
```

Comment:
Typically guard types are just dropped to invoke their "disable" functionality, and any direct methods on them like this are actually to disable the _guard_, not to disable or release whatever the guard is holding on to (e.g. see [`disarm`](https://docs.rs/futures-intrusive/0.3.1/futures_intrusive/sync/struct.GenericSemaphoreReleaser.html#method.disarm) on the `futures_intrusive` semaphore releaser type). Given that, I'd suggest just going back to only a `Drop` implementation and doing `fail_point_guards.clear()` in the test runner to drop all the fail point guards rather than iterating over them and manually disabling their failpoints.

Reply:
Ah makes sense, I didn't realize `clear` would call `drop` -- I'll revert.